packages/playground/cli/src/cli.ts

```diff
@@ -348,6 +348,11 @@ async function run() {
 				process.exit(0);
 			} else {
 				logger.log(`WordPress is running on ${absoluteUrl}`);
+				if (args.login) {
```

Comment:
I'm starting to think we should just ship a WordPress plugin to log the user in the first time they interact with the site. It could go like this:

1. The `login` step would install that mu-plugin. Or maybe the plugin would always be there and `login` would just  set some constant to activate it?
2. The plugin would check for a cookie like `wordpress_user_was_auto_logged`. If it's present, we're done. If it's missing, set it and also log the user in.

This should work in all runtimes (Studio, Playground CLI, WP-NOW, Playground webapp) without any runtime-specific loginc.

Reply:
Claude gave me this. Coming from AI, there's likely some problem with that code, but it seems like a good starting point:

```ts
<?php
/*
Plugin Name: Auto Admin Login
Description: Automatically logs in as admin if a specific cookie is not present
Version: 1.0
Author: Your Name
*/

// Function to check cookie and perform auto-login
function auto_admin_login() {
    $cookie_name = 'wordpress_user_was_auto_logged';

    // Check if the cookie exists
    if (isset($_COOKIE[$cookie_name])) {
        return;
    }

    // Cookie doesn't exist, set it
    setcookie($cookie_name, '1', time() + (86400 * 30), '/'); // Cookie expires in 30 days

    // Get the admin user
    $admin_user = get_user_by('login', 'admin');

    // If admin user exists, log them in
    if ($admin_user) {
        wp_set_current_user($admin_user->ID);
        wp_set_auth_cookie($admin_user->ID);
        do_action('wp_login', $admin_user->user_login, $admin_user);
    }
}

// Hook the function to run on WordPress init
add_action('init', 'auto_admin_login');
```

Reply:
Yeah, good idea. However, I'm concerned about using a plugin in Studio's case, since we'd need to filter it out when exporting the site.

Reply:
Oh I don't want to pollute the exports, filtering files out just doesn't work as a long term strategy. The login plugin would live in `/internal/shared/mu-plugins` with the rest of the platform-level mu-plugins to keep the site clean.

Reply:
> The login plugin would live in `/internal/shared/mu-plugins` with the rest of the platform-level mu-plugins to keep the site clean.

Ah, ok, I didn't know about this approach 😅.

Still, I wonder about how we could make this plugin work so the user can decide when it should auto-login and when not. If I understand correctly the code shared [here](https://github.com/WordPress/wordpress-playground/pull/1753#discussion_r1755631258), the `init` hook will make that all requests are logged, right?

Reply:
@fluiddot here's my thinking:

* The runtime (Studio, Playground webapp etc.) decides whether or not the plugin is installed at any given time. The plugin is easy to install and easy to remove – it's just a php file.
* When the plugin is in place, it automatically logs the user in as an admin, or as another user when a constant / env variable like `AUTOLOGIN_USERNAME` is present.
* Autologin sets auth cookies valid for a year, and an extra cookie called, say, `autologin_performed`
* If the `autologin_performed` cookie is already set, no autologin happens.

This way the user will remain logged in until they explicitly log out, at which point they'll remain logged out until they explicitly log in. A private browsing session will still default to the logged in state.

There could also be a "querystring mode" where autologin is only performed when `$_GET['playground-autologin']` is present. This would give Studio a link-based control while still allowing the Playground webapp and CLI to default to autologin.

What do you think?

Reply:
The approach looks good to me, thanks @adamziel ! I'll try to work on this in the next few days. Note that the [original issue](https://github.com/Automattic/studio/issues/387) lowered its priority, so I might delay updating the PR.

> There could also be a "querystring mode" where autologin is only performed when `$_GET['playground-autologin']` is present. This would give Studio a link-based control while still allowing the Playground webapp and CLI to default to autologin.

At least for the Studio app, and probably when using node clients, this will be necessary to ensure that API requests to the site are not auto-logged. This is important when using, for instance, API keys as referenced in https://github.com/Automattic/studio/issues/387.

Reply:
A check like `if(!$ajax && !$rest_api) could also help solve for the API requests. Anyway, whatever's most useful here.